<launch>

  <!-- set /global_example parameter -->  
  <group ns="renov_up_level">

    <param name="open_climb_flag" value="0" />

    <param name="open_hold_flag" value="0"/>

    <param name="open_rotation_flag" value="0"/>



    <param name="climb_kp" value="10000"/>
    <param name="climb_ki" value="0.1"/>
    <param name="climb_kd" value="0.1"/>

    <param name="hold_kp" value="30000"/>
    <param name="hold_ki" value="0.1"/>
    <param name="hold_kd" value="0.1"/>

    <param name="rotation_kp" value="15"/>
    <param name="rotation_ki" value="0.1"/>
    <param name="rotation_kd" value="0.1"/>

    <param name="stand_bar_flex_distance" value="0.437"/>
    <param name="light_scan_to_top_distance" value="0.810"/>

    <param name="rotation_joint_line_equation_k" value="0.0035634"/>
    <param name="rotation_joint_line_equation_b" value="-158.2"/>

    <!--set /climb_pkg_para/close all 3dof driver-->
    <param name="close_all_3dof_climb_driver_flag" value="0"/>

    <param name="aubo_painting_opreating_over" value="0" />
    <param name="climb_distance_tracking_over" value="0" />
    <param name="rotation_distance_tracking_over" value="0" />
    <param name="hold_distance_tracking_over" value="0" />

    <!--set /climb_pkg_para/enable_control_stand_bar 1 enable 2 disable-->
    <param name="enable_control_stand_bar" value="0" />
    <!--set /climb_pkg_para/enable_control_rotation 1 enable 2 disable-->
    <param name="enable_control_rotation" value="0"/>
    <!--set /climb_pkg_para/enable_climb_control-->
    <param name="enable_climb_control" value="0"/>
    <!-- close driver second -->
    <param name="enable_second_control_stand_bar" value="0" />

    <param name="enable_second_climb_control" value="0"/>


    <param name="enable_third_stand_bar" value="0"/>
    <!-- 1,open,0close -->
    <param name="open_hold_to_ceil_flag" value="0"/>
    <param name="rotation_homing_abs_encode_data" value="44396"/>


    <param name="velocity_control_stand_bar" value="1000" />
    <!--set /climb_pkg_para/enable_control_rotation-->
    <param name="velocity_control_rotation" value="2000"/>
    <!--set /climb_pkg_para/enable_climb_control-->
    <param name="velocity_climb_control" value="1000"/>

    <!--set /climb_pkg_para/enable_control_stand_bar velocity-0.05/8=0.00625m/s 0.0-0.3-max neg-down pos-up-->
    <param name="distance_control_stand_bar" value="0" />
    <!--set /climb_pkg_para/enable_control_rotation -3.14-0.8 max pos-anticlockwise neg-clokwise-->
    <param name="rad_control_rotation" value="0"/>
    <!--set /climb_pkg_para/enable_climb_control 0-2.5m max pos-up neg-down-->
    <param name="distance_climb_control" value="0"/>    

    <node pkg="painting_robot_demo" type="climb_bar_startup.py" name="climb_bar_startup" output="screen" >
    </node>
    
  </group>
  <include file="$(find painting_robot_demo)/launch/climb_flex_bar_holding.launch">
  </include>
  <include file="$(find painting_robot_demo)/launch/climb_flex_bar_homing.launch">
  </include>
  
</launch>

